lib/sftp_client/operation_util.ex

```diff
@@ -52,6 +52,10 @@ defmodule SFTPClient.OperationUtil do
     %InvalidOptionError{key: key, value: load_opt_value(value), reason: reason}
   end
 
+  def handle_error({:error, reason}) do
+    %ConnError{message: to_string(reason)}
+  end
```

Comment:
Which particular case is this supposed to handle? As far as I can see here, this would only handle errors with following format and not the one you described:

```elixir
{:error, {:error, reason}}
```

The docs state the following:

> The atom() value is formed from the sftp error codes in the protocol-level responses as defined in draft-ietf-secsh-filexfer-13 section 9.1. The codes are named as SSH_FX_* which are transformed into lowercase of the star-part. E.g. the error code SSH_FX_NO_SUCH_FILE will cause the reason() to be no_such_file.

This is handled by:

```elixir
  def handle_error(reason) when is_atom(reason) do
    %OperationError{reason: reason}
  end
``` 

> The string() reason is the error information from the server in case of an exit-signal. If that information is empty, the reason is the exit signal name.

This is handled by:

```elixir
  def handle_error(message) do
    %ConnError{message: to_string(message)}
  end
```

> The tuple() reason are other errors like for example {exit_status,1}.

This is currently unhandled _except_ for this particular case:

```elixir
  def handle_error({:eoptions, {{key, value}, reason}}) do
    %InvalidOptionError{key: key, value: load_opt_value(value), reason: reason}
  end
```

So, I guess the change should better be to intercept all kinds of tuples:

```suggestion
  def handle_error(reason) when is_tuple(reason) do
    %ConnError{message: inspect(reason)}
  end
```

Reply:
Sorry for the late response but had to return in context for this!
Yeah, this solution is way more generic and better than my particular case - which was simply one I encountered using the library.
I can edit the PR with your code, or just close this